AppDB/appscale/datastore/cassandra_env/large_batch.py

```diff
@@ -214,3 +224,214 @@ def claim(self):
+
+    Args:
+      txid_hash: A byte array identifying the transaction.
+    Returns:
+      A Cassandra result for the batch entry.
+    """
+    statement = self._get_prepared("""
+      SELECT applied, op_id FROM batch_status
+      WHERE txid_hash = ?
+    """)
+    bound_statement = statement.bind((txid_hash,))
+    bound_statement.consistency_level = ConsistencyLevel.SERIAL
+    bound_statement.retry_policy = BASIC_RETRIES
+    results = yield self._tornado_cassandra.execute(bound_statement)
+    try:
+      raise gen.Return(results[0])
+    except IndexError:
+      raise BatchNotFound('Batch not found')
+
+  @gen.coroutine
+  def _insert(self, txid_hash, op_id):
+    """ Claims the large batch.
+
+    Args:
+      txid_hash: A byte array identifying the transaction.
+      op_id: A uuid4 specifying the process ID.
+    """
+    statement = self._get_prepared("""
+      INSERT INTO batch_status (txid_hash, applied, op_id)
+      VALUES (?, ?, ?)
+      IF NOT EXISTS
+    """)
+    bound_statement = statement.bind((txid_hash, False, op_id))
+    bound_statement.retry_policy = NO_RETRIES
+    results = yield self._tornado_cassandra.execute(bound_statement)
+    if not results[0].applied:
+      raise BatchNotOwned('Another process started applying the transaction')
+
+  @gen.coroutine
+  def _select_mutations(self, txid):
+    """ Fetches a list of the mutations for the batch.
+
+    Args:
+      txid: An integer specifying a transaction ID.
+    Returns:
+      An iterator of Cassandra results.
+    """
+    statement = self._get_prepared("""
+      SELECT old_value, new_value FROM batches
+      WHERE app = ? AND transaction = ?
+    """)
+    bound_statement = statement.bind((self.project_id, txid))
+    bound_statement.retry_policy = BASIC_RETRIES
+    results = yield self._tornado_cassandra.execute(bound_statement)
+    raise gen.Return(results)
+
+  @gen.coroutine
+  def _apply_mutations(self, txid, composite_indexes):
+    """ Applies all the mutations in the batch.
+
+    Args:
+      txid: An integer specifying a transaction ID.
+      composite_indexes: A list of CompositeIndex objects.
+    """
+    results = yield self._select_mutations(txid)
+    futures = []
+    for result in results:
+      old_entity = result.old_value
+      if old_entity is not None:
+        old_entity = entity_pb.EntityProto(old_entity)
+
+      new_entity = result.new_value
+
+      if new_entity is None:
+        mutations = deletions_for_entity(old_entity, composite_indexes)
+      else:
+        new_entity = entity_pb.EntityProto(new_entity)
+        mutations = mutations_for_entity(new_entity, txid, old_entity,
+                                         composite_indexes)
+
+      statements_and_params = self._db_access.statements_for_mutations(
+        mutations, txid)
+      for statement, params in statements_and_params:
+        futures.append(self._tornado_cassandra.execute(statement, params))
+
+    for future in futures:
+      yield future
```

Comment:
Isn't it better to yield all futures together?
```
yield futures
```

Reply:
Cool. I did not know you could yield a list of futures. That is a convenient syntax.